yellowbrick/features/jointplot.py

```diff
@@ -14,308 +14,386 @@
+        self._yhax.grid(False, axis='x')
 
-            # If X is a data frame, get the columns off it.
-            if is_dataframe(X):
-                self.feature = X.columns
+    def fit(self, X, y=None):
+        """
+        Fits the JointPlot, creating a correlative visualization between the columns
+        specified during initialization and the data and target passed into fit:
 
-            else:
-                self.feature = ['x']
+            - If self.columns is None then X and y must both be specified as 1D arrays
+              or X must be a 2D array with only 2 columns.
+            - If self.columns is a single int or str, that column is selected to be
+              visualized against the target y.
+            - If self.columns is two ints or strs, those coumns are visualized against
+              each other. If y is specified then it is used to color the points.
 
-        # Handle the target name if it is None.
-        if self.target is None:
-            self.target = ['y']
+        This is the main entry point into the joint plot visualization.
 
-        self.draw(X, y, **kwargs)
-        return self
+        Parameters
+        ----------
+        X : array-like
+            An array-like object of either 1 or 2 dimensions depending on self.columns.
+            Usually this is a 2D table with shape (n, m)
 
-    def draw(self, X, y, **kwargs):
-        """
-        Sets up the layout for the joint plot draw calls ``draw_joint`` and
-        ``draw_xy`` to render the visualizations.
+        y : array-like, default: None
+            An vector or 1D array that has the same length as X. May be used to either
+            directly plot data or to color data points.
         """
-        fig = plt.gcf()
-        gs = plt.GridSpec(self.ratio + 1, self.ratio + 1)
+        # Convert python objects to numpy arrays
+        if isinstance(X, (list, tuple)):
+            X = np.array(X)
 
-        #Set up the 3 axes objects
-        joint_ax = fig.add_subplot(gs[1:, :-1])
-        x_ax = fig.add_subplot(gs[0, :-1], sharex=joint_ax)
-        y_ax = fig.add_subplot(gs[1:, -1], sharey=joint_ax)
+        if y is not None and isinstance(y, (list, tuple)):
+            y = np.array(y)
 
-        fig.tight_layout()
-        fig.subplots_adjust(hspace=self.space, wspace=self.space)
+        # Case where no columns are specified
+        if self.columns is None:
+            if (y is None and (X.ndim != 2 or X.shape[1] != 2)) or (y is not None and (X.ndim != 1 or y.ndim != 1)):
+                raise YellowbrickValueError((
+                    "when self.columns is None specify either X and y as 1D arrays "
+                    "or X as a matrix with 2 columns"
+                ))
 
-        self.fig = fig
-        self.joint_ax = joint_ax
-        self.x_ax = x_ax
-        self.y_ax = y_ax
-        self._ax = joint_ax
+            if y is None:
+                # Draw the fist column as x and the second column as y
```

Comment:
Small typo: fist=>first

Reply:
But a ✊  column would be cool, sort of a "power to the people" kind of thing.